### PR TITLE
infernalis: ceph-disk: use blkid instead of sgdisk -i

### PR DESCRIPTION
**Pull request by dachary**
http://tracker.ceph.com/issues/14145

### CODE REVIEW

**Comment by dachary** [comment]
<pre>
cw ceph-qa-suite --verbose --ceph-qa-suite-git-url http://github.com/ceph/ceph-qa-suite --suite-branch infernalis --ceph-git-url http://github.com/dachary/ceph --ceph wip-14145-infernalis --suite ceph-disk --upload
</pre>

* *running* http://167.114.233.29:8081/ubuntu-2015-12-21_11:20:05-ceph-disk-wip-14145-infernalis---basic-openstack/

File: src/ceph-disk [hunk 1]
```diff
@@ -1217,14 +1219,30 @@ def update_partition(dev, description):
...
+            _check_output(['partprobe', dev])
+            partprobe_ok = True
+            break
+        except subprocess.CalledProcessError as e:
+            error = e.output
+            if ('unable to inform the kernel' not in error and
+                'Device or resource busy' not in error):
+                raise
+            LOG.debug('partprobe %s failed : %s (ignored, waiting 60s)' % (dev, error))
+            time.sleep(60)
```
Comment (fenderle) [inline src/ceph-disk]: I found to have exactly that problem (Device or resource busy) using ceph-disk prepare. This fixes it for me, though the 60s delay is a bit high (I changed that to 10s and it worked fine)

**Comment by dachary** [comment]
<pre>
cw ceph-qa-suite --verbose --ceph-qa-suite-git-url http://github.com/ceph/ceph-qa-suite --suite-branch infernalis --ceph-git-url http://github.com/dachary/ceph --ceph wip-14145-infernalis --suite ceph-disk --upload
</pre>

* *running* http://167.114.238.126:8081/ubuntu-2016-01-06_13:36:00-ceph-disk-wip-14145-infernalis---basic-openstack

**Comment by dachary** [comment]
<pre>
cw ceph-qa-suite --verbose --ceph-qa-suite-git-url http://github.com/ceph/ceph-qa-suite --suite-branch infernalis --ceph-git-url http://github.com/dachary/ceph --ceph wip-14145-infernalis --suite ceph-disk --upload
</pre>

* *running* http://167.114.238.126:8081/ubuntu-2016-01-06_22:43:05-ceph-disk-wip-14145-infernalis---basic-openstack/

**Comment by dachary** [comment]
@liewegas this backport is not a clean cherry pick because deactivate / destroy did not exist back in infernalis but is otherwise the same. The 5798d1e commit has been added for the same reason. It passed the ceph-disk suite ( see http://teuthology-logs.public.ceph.com/ubuntu-2016-01-06_22%3a43%3a05-ceph-disk-wip-14145-infernalis---basic-openstack/ ). Does it look ready to merge ?

**Comment by liewegas** [comment]
:+1:

**Pull request merged**